weavermetadata/metadata.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package weavermetadata define the structure of the metadata supported by weaver.
```

Comment:
Reword:

Package weavermetadata provides support for the propagation of metadata information from a component method caller to the callee. The metadata is propagated to the callee even if the caller and callee are not colocated in the same process.

The metadata is a map from string to string stored in context.Context. The map can be added to a context by calling NewContext.

... include example here ...

The metadata map can be extracted from a context by calling FromContext:

... include example here ...

Reply:
Done